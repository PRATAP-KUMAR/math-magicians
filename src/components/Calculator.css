.calc-result {
  justify-content: right;
  color: #fff;
  background-color: rgb(77, 71, 71);
  height: 50px;
  align-items: center;
  display: flex;
  padding-right: 10px;
  font-size: 30px;
}

.buttons-grid {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-auto-rows: minmax(60px, auto);
}

.button {
  background-color: #b4bebb;
  font-size: 20px;
}

.button:nth-child(4n) {
  background-color: orange;
}

.button:hover {
  background-color: #f45678;
}

.button:nth-child(4n):hover {
  background-color: blue;
}
