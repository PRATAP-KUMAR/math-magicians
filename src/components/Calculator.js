import React from 'react';
import './calculator.css';
import calculate from '../logic/calculate';
import NavLinks from './NavLinks';

class Calculator extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      operation: null,
    };
  }

  handleInput = (e) => {
    const myVar = e.target.textContent;
    const obj = calculate(this.state, myVar);
    this.setState(obj);
  };

  render() {
    const { total, next, operation } = this.state;
    return (
      <>
        <NavLinks />
        <div className="calc-container">
          <div className="some-math-text">
            Lets do some math
          </div>
          <div className="calc">
            <div className="calc-result">{next || operation || total || 0}</div>
            <div className="buttons-grid">
              <button type="button" className="button" value="AC" onClick={(e) => this.handleInput(e)}>AC</button>
              <button type="button" className="button" value="+/-" onClick={(e) => this.handleInput(e)}>+/-</button>
              <button type="button" className="button" value="%" onClick={(e) => this.handleInput(e)}>%</button>
              <button type="button" className="button" value="&divide;" onClick={(e) => this.handleInput(e)}>&divide;</button>
              <button type="button" className="button" value="7" onClick={(e) => this.handleInput(e)}>7</button>
              <button type="button" className="button" value="8" onClick={(e) => this.handleInput(e)}>8</button>
              <button type="button" className="button" value="9" onClick={(e) => this.handleInput(e)}>9</button>
              <button type="button" className="button" value="x" onClick={(e) => this.handleInput(e)}>x</button>
              <button type="button" className="button" value="4" onClick={(e) => this.handleInput(e)}>4</button>
              <button type="button" className="button" value="5" onClick={(e) => this.handleInput(e)}>5</button>
              <button type="button" className="button" value="6" onClick={(e) => this.handleInput(e)}>6</button>
              <button type="button" className="button" value="-" onClick={(e) => this.handleInput(e)}>-</button>
              <button type="button" className="button" value="1" onClick={(e) => this.handleInput(e)}>1</button>
              <button type="button" className="button" value="2" onClick={(e) => this.handleInput(e)}>2</button>
              <button type="button" className="button" value="3" onClick={(e) => this.handleInput(e)}>3</button>
              <button type="button" className="button" value="+" onClick={(e) => this.handleInput(e)}>+</button>
              <button type="button" className="button"> </button>
              <button type="button" className="button" value="0" onClick={(e) => this.handleInput(e)}>0</button>
              <button type="button" className="button" value="." onClick={(e) => this.handleInput(e)}>.</button>
              <button type="button" className="button" value="=" onClick={(e) => this.handleInput(e)}>=</button>
            </div>
          </div>
        </div>
      </>
    );
  }
}

export default Calculator;
