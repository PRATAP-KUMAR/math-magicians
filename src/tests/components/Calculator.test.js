import { MemoryRouter } from 'react-router-dom';
import { render, fireEvent, screen } from '@testing-library/react';

import Calculator from '../../components/Calculator';

describe('Calculator Component Test', () => {
  test('test for Calculator', () => {
    render(<MemoryRouter><Calculator /></MemoryRouter>);
    fireEvent.click(screen.getByText('2'));
    fireEvent.click(screen.getByText('+/-'));
    fireEvent.click(screen.getByText('x'));
    fireEvent.click(screen.getByText('4'));
    fireEvent.click(screen.getByText('='));
    expect(screen.getByText('-8')).toBeInTheDocument();
    expect(screen.getByText('-8')).toMatchSnapshot();
  });
});
