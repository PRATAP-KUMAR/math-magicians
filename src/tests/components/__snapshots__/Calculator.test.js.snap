// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Component Test test for Calculator 1`] = `
<div
  class="calc-result"
>
  -8
</div>
`;
